frontend/src/components/App.tsx

```diff
@@ -1,71 +1,119 @@
 import * as PageConstants from "./pages/page_constants";
 import * as React from "react";
 import AppBar from "./appbar/AppBar";
-import { BrowserRouter } from "react-router-dom";
+import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";
 import clsx from "clsx";
 import CssBaseline from "@material-ui/core/CssBaseline";
+import { defaultServices } from "./contexts/default_services";
 import Drawer from "./drawer/Drawer";
 import { Explore } from "./pages/explore/Explore";
 import { hot } from "react-hot-loader";
 import { Login } from "./pages/login/Login";
 import Profile from "./pages/profile/Profile";
-import { Route } from "react-router-dom";
-import { Switch } from "react-router-dom";
-import { ServiceContext } from "./contexts/contexts";
-import { defaultServices } from "./contexts/default_services";
+import { ServiceContext, UserLoginStatusContext } from "./contexts/contexts";
+import { userLoginServices } from "./contexts/user_login_services";
 import { withStyles } from "@material-ui/core/styles";
 import { YourClubs } from "./pages/your_clubs/YourClubs";
 
-interface AppProps {}
+function App() {
+  // Construct dependencies to determine status of user login.
+  const userLoginStatusService = userLoginServices.loginStatusHandlerService;
+  const [navigationDrawerOpen, setNavigationDrawerOpen] = React.useState<boolean>(false);
+  const [userLoggedIn, setUserLoggedIn] = React.useState<boolean|undefined>(undefined);
```

Comment:
nit: Is there a reason you need to distinguish between undefined and false? Or could you have it just default to false?